src/preferences/PreferencesManager.js

```diff
@@ -44,29 +44,63 @@ define(function (require, exports, module) {
         Strings                 = require("strings"),
         PreferencesImpl         = require("preferences/PreferencesImpl"),
         _                       = require("thirdparty/lodash");
-    
+
     /**
      * The local storage ID
      * @const
      * @type {string}
      */
     var PREFERENCES_CLIENT_ID = "com.adobe.brackets.preferences";
-    
+
     /**
      * The prefix used in the generated client ID
      * @const
      * @type {string}
      */
     var CLIENT_ID_PREFIX = "com.adobe.brackets.";
-    
+
     // Private Properties
     var preferencesKey,
         prefStorage,
         persistentStorage,
         extensionPaths,
         doLoadPreferences   = false;
-    
-    
+
+    var memoryStorage = {
+        _items: {},
+        getItem: function(key) {
+            return memoryStorage._items[key];
+        },
+        setItem: function(key, value) {
+            // Mimic localStorage string storage
+            value = "" + value;
+            memoryStorage._items[key] = value;
+        },
+        removeItem: function(key) {
+            delete memoryStorage._item[key];
+        },
+        clear: function() {
+            memoryStorage._items = {};
+        }
+    };
+
+    var localStorage = (function(window) {
+        var localStorage;
+
+        try {
+            localStorage = window.localStorage;
+            if(typeof window.localStorage === 'undefined') {
+                localStorage = memoryStorage;
+            }
+        } catch(e) {
+            console.warn("Unable to use localStorage in Brackets with: ", e);
+            localStorage = memoryStorage;
+        } finally {
+            return localStorage;
+        }
+    }(window));
```

Comment:
I wonder if you should actually be doing this sooner, specifically in https://github.com/mozilla/brackets/blob/master/src/utils/Compatibility.js.  That way, if some other part of the code starts using `localStorage`, we aren't going to have an order bug.